plugins/pur/core/src/main/java/com/pentaho/di/services/PentahoDiPlugin.java

```diff
@@ -213,39 +212,39 @@ public PentahoDiPlugin.PurRepositoryPluginApiPurge.PathIdPurge setPathid( String
         return new PentahoDiPlugin.PurRepositoryPluginApiPurge.PathIdPurge( _client, copyUriBuilder, copyMap );
       }
 
-      public <T> T postMultipartFormDataAs( Object input, com.sun.jersey.api.client.GenericType<T> returnType ) {
+      public <T> T postMultipartFormDataAs( Object input, javax.ws.rs.core.GenericType<T> returnType ) {
         UriBuilder localUriBuilder = _uriBuilder.clone();
-        com.sun.jersey.api.client.WebResource resource =
-            _client.resource( localUriBuilder.buildFromMap( _templateAndMatrixParameterValues ) );
-        com.sun.jersey.api.client.WebResource.Builder resourceBuilder = resource.getRequestBuilder();
+        javax.ws.rs.client.WebTarget resource =
+            _client.target( localUriBuilder.buildFromMap( _templateAndMatrixParameterValues ) );
+        javax.ws.rs.client.Invocation.Builder resourceBuilder = resource.request();
         resourceBuilder = resourceBuilder.accept( "*/*" );
-        resourceBuilder = resourceBuilder.type( "multipart/form-data" );
-        com.sun.jersey.api.client.ClientResponse response;
-        response = resourceBuilder.method( "POST", com.sun.jersey.api.client.ClientResponse.class, input );
+        resourceBuilder = resourceBuilder.header( "Content-Type" , "multipart/form-data" );
+        javax.ws.rs.core.Response response;
+        response = resourceBuilder.method( "POST" );
         if ( response.getStatus() >= 400 ) {
           throw new PentahoDiPlugin.WebApplicationExceptionMessage( Response
-              .status( response.getClientResponseStatus() ).build() );
+              .status( response.getStatus() ).build() );
         }
-        return response.getEntity( returnType );
+        return (T) response.getEntity();
       }
 
       public <T> T postMultipartFormDataAs( Object input, Class<T> returnType ) {
         UriBuilder localUriBuilder = _uriBuilder.clone();
-        com.sun.jersey.api.client.WebResource resource =
-            _client.resource( localUriBuilder.buildFromMap( _templateAndMatrixParameterValues ) );
-        com.sun.jersey.api.client.WebResource.Builder resourceBuilder = resource.getRequestBuilder();
+        javax.ws.rs.client.WebTarget resource =
+            _client.target( localUriBuilder.buildFromMap( _templateAndMatrixParameterValues ) );
+        javax.ws.rs.client.Invocation.Builder resourceBuilder = resource.request();
         resourceBuilder = resourceBuilder.accept( "*/*" );
-        resourceBuilder = resourceBuilder.type( "multipart/form-data" );
-        com.sun.jersey.api.client.ClientResponse response;
-        response = resourceBuilder.method( "POST", com.sun.jersey.api.client.ClientResponse.class, input );
-        if ( !com.sun.jersey.api.client.ClientResponse.class.isAssignableFrom( returnType ) ) {
+        resourceBuilder = resourceBuilder.header( "Content-Type" , "multipart/form-data" );
+        javax.ws.rs.core.Response response;
+        response = resourceBuilder.method( "POST" );
```

Comment:
Did you forget to mention Response.class as second argument as I see it for other occurrences (like in 362 line)?